os/HowToUseDutils.md

```diff
@@ -0,0 +1,189 @@
+# How to use dutils.sh
+```
+cd <path>/os
+```
+
+## To run TOOLCHAIN,
+
+```
+When user has not installed toolchain, but only using the docker container, then there needs to be a method to run toolchain commands like nm, readelf, addr2line for debugging purpose.
+
+The user can call dutils.sh with new "tool" command, followed by the actual toolchain command to be run.
+
+./dutils.sh tool nm build/output/bin/tinyara.axf
+./dutils.sh tool addr2line -e build/output/bin/tinyara.axf 0x0123abcd
+
+
+The toolchaing commands can also be run from the dutils as below:
+
+
+> Run ./dutils.sh
+> Press 2
+> Select the desired option
+```
+
+### Example
+```
+test@Focal:~/Desktop/product_Refrigerator/.tizenrt/os$ sudo ./dutils.sh 
+Check Docker Image
+Docker Image Version : tizenrt/tizenrt:1.5.6
+======================================================
+  "Select Option"
+======================================================
+  "1. Run TRAP Debug tool"
+  "2. Run Toolchain Command"
+  "h. Help"
+  "x. Exit"
+======================================================
+2
+make: Entering directory '/home/test/Desktop/product_Refrigerator/.tizenrt/os/tools'
+chmod 755 ../makeinfo.sh
+make: Leaving directory '/home/test/Desktop/product_Refrigerator/.tizenrt/os/tools'
+======================================================
+  "Select Binary for Toolchain Command"
+======================================================
+  "1. Kernel"
+  "2. App1"
+  "3. App2"
+  "4. Common"
+  "5. Enter manually"
+  "x. Exit"
+======================================================
+1
+======================================================
+  "Select Toolchain Command"
+======================================================
+  "1. nm"
+  "2. addr2line"
+  "3. Enter manually"
+  "x. Exit"
+======================================================
+2
+Enter address: 
+0e07844c
+Executing: arm-none-eabi-addr2line -e ../build/output/bin/tinyara.axf 0e07844c  
+net.c:?
+>> Output is stored in dutils_output_addr2line_2024.09.06-14.33.16.txt
+
+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+```
+
+## To run TRAP tool,
+
+```
+> copy crash logs, if any, in the repository
```

Comment:
in the repository  ==> in the repository

Provide some information about where to copy the logs. Is it home folder, or somewhere else.